perf: reanimated 3 upgrade

hannojg opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

This PR upgrades reanimated to version 3. It's based on https://github.com/Expensify/App/pull/11550 by @marcaaron since his PR removes any reanimated v1 usage (which rea v3 dropped support for). [This PR has since landed on `main`, so that's good!]

This PR is an effort toward enabling fabric architecture.

### Fixed Issues
<!---
Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing.
Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the issue; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
Related https://github.com/Expensify/App/issues/8503

### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Crop and edit your avatar
2. Use the left hand nav bar to navigate to reports

### PR Review Checklist
<!--
This is a checklist for PR authors & reviewers. Please make sure to complete all tasks and check them off once you do, or else Expensify has the right not to merge your PR!
-->
#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
3. Upload an image via copy paste
4. Verify a modal appears displaying a preview of that image
--->

1. Crop and edit your avatar
2. Use the left hand nav bar to navigate to reports

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->
<img width="1519" alt="Screen Shot 2022-10-20 at 1 29 25 PM" src="https://user-images.githubusercontent.com/2838819/197040695-b9d520f2-0e6c-4592-8fbd-63f1e28942e9.png">

<img width="1519" alt="Screen Shot 2022-10-20 at 1 29 45 PM" src="https://user-images.githubusercontent.com/2838819/197040700-e41ff3c4-2ebd-4dad-a205-61ee832e28c9.png">

#### Mobile Web - Chrome
<img width="1519" alt="Screen Shot 2022-10-20 at 1 30 14 PM" src="https://user-images.githubusercontent.com/2838819/197040757-1209259a-68b5-480c-bfe4-3621ee5d1fb1.png">
<img width="1519" alt="Screen Shot 2022-10-20 at 1 30 34 PM" src="https://user-images.githubusercontent.com/2838819/197040760-f5f006c4-8020-4a71-b769-f955d4b0054a.png">



#### Mobile Web - Safari
<!-- Insert screenshots of your changes on the web platform (from safari mobile browser)-->

<img width="564" alt="Screen Shot 2022-10-20 at 1 35 26 PM" src="https://user-images.githubusercontent.com/2838819/197041496-ff792e4b-fcf1-4b8e-9aef-f67b5da44d0f.png">
<img width="564" alt="Screen Shot 2022-10-20 at 1 35 11 PM" src="https://user-images.githubusercontent.com/2838819/197041508-6a431a0f-9838-4f43-be5e-e88e4867ba0c.png">


#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->
<img width="1312" alt="Screen Shot 2022-10-20 at 1 32 34 PM" src="https://user-images.githubusercontent.com/2838819/197041043-ee258927-6e95-4fd1-b909-c2d0bc3037fb.png">

<img width="1312" alt="Screen Shot 2022-10-20 at 1 32 55 PM" src="https://user-images.githubusercontent.com/2838819/197041048-4ac37c45-ec3f-4e7e-9dcd-d983c52fd4f9.png">

#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->
<img width="564" alt="Screen Shot 2022-10-20 at 1 15 28 PM" src="https://user-images.githubusercontent.com/2838819/197038440-47851994-72d0-48f1-ab8b-028a851efad4.png">
<img width="564" alt="Screen Shot 2022-10-20 at 1 15 49 PM" src="https://user-images.githubusercontent.com/2838819/197038451-1d905ef9-a08c-4038-9134-5a4b20ee61f0.png">

#### Android
<img width="747" alt="Screen Shot 2022-10-20 at 1 20 21 PM" src="https://user-images.githubusercontent.com/2838819/197038889-210727c6-8d9d-465b-a784-e0af9895a886.png">
<img width="747" alt="Screen Shot 2022-10-20 at 1 20 59 PM" src="https://user-images.githubusercontent.com/2838819/197038891-62f5c728-1d13-488a-ba6a-3001c8f42e3e.png">

hannojg committed:
bump REA package

hannojg committed:
android: remove obsolete REA configuration

hannojg committed:
Merge branch 'main' of github.com:margelo/expensify-app-fork into perf/rea-3-upgrade

hannojg committed:
Merge branch 'main' of github.com:margelo/expensify-app-fork into perf/rea-3-upgrade

hannojg commented:
Update:

After talking to the team at react-navigation it appears that we have to use force install, or wait until reanimated v3 is officially released:
https://github.com/react-navigation/react-navigation/issues/10909#issuecomment-1266955497

hannojg committed:
Merge branch 'main' of github.com:margelo/expensify-app-fork into perf/rea-3-upgrade

hannojg committed:
pod install

hannojg committed:
Merge branch 'main' of github.com:margelo/expensify-app-fork into perf/rea-3-upgrade

hannojg committed:
temp: patch package to fix reanimated v3-rc installation issue

hannojg commented:
I just [added a fix](https://github.com/Expensify/App/pull/11567/commits/25cf51c3171903838007f9cff3ee531c2cd5a316) with `patch-package` for the installation issue, which we can use until REA v3 is officially released.
Are we cool with that?

roryabraham commented:
Yes, using `patch-package` is fine 👍🏼 Let's get this PR polished up and over the finish line

hannojg committed:
fix tests

Santhosh-Sellavel commented:
@hannojg Are there any test steps here, Does something need to compare?

cc: @Beamanator

hannojg committed:
Revert "pod install"

This reverts commit d8a07bfae8c443b9c2ef9af29f83e7dfefa9156a.

hannojg commented:
@Santhosh-Sellavel this is just some internal library update. Reanimated isn't used too much around the code base. Most noticeably it's used for the drawer (reports list/ SideBarLinks and the actual Report View). So any interaction with the drawer maybe should be paid attention to.

However, really opening the app and it not crashing should be test-steps enough for this change 😄

Beamanator commented:
Looks like maybe image / avatar cropping as well?

AndrewGable reviewed (dismissed):
Code looks good - I'm going to edit the PR description to make sure it's "kosher" then have @roryabraham review.

Santhosh-Sellavel commented:
@AndrewGable 
> 1. Crop and edit your avatar

Got this

> 2. Use the left hand nav bar to navigate to reports

Sorry, I don't get what you mean here. Can you please elaborate? Thanks!

AndrewGable commented:
Click in and out of chats using the "home page" and LHN on web.

<img width="747" alt="Screen Shot 2022-10-20 at 1 20 21 PM" src="https://user-images.githubusercontent.com/2838819/197039248-d6f75dd4-f72d-4ded-b9d8-54425445a0b1.png">

Santhosh-Sellavel commented:
Both are separate actions, sequencing them as 1 & 2 got me confused I mistook them as Test Steps, thanks for the clarification!

roryabraham commented on patches/@react-navigation+drawer+6.5.0.patch:
```diff
@@ -0,0 +1,13 @@
+     "react-native": "*",
+     "react-native-gesture-handler": ">= 1.0.0",
+-    "react-native-reanimated": ">= 1.0.0",
++    "react-native-reanimated": "*",
```
Just FYI, I was curious whether the warning I linked while testing this would be fixed by upgrading `@react-navigation/drawer`, so I published a new version of our fork with this change (gitpkg and patch-package didn't seem to work together).

So you can just use `"@react-navigation/drawer": "git@github.com:Expensify/react-navigation.git#react-navigation-drawer-v6.5.0-alpha1-gitpkg"` in the package.json instead of using `patch-package` here.

AndrewGable commented:
@hannojg - Any ideas on [this warning](https://github.com/Expensify/App/pull/11567#issuecomment-1286111647)? It seems like a blocker to merge the PR, if it's harmless, maybe we can ignore it?

hannojg commented:
Let me look into it! With Rory's fix I can also remove the patch

hannojg commented:
Hm, it seems to me like a race condition where the native side starts sending events to JS side, while the JS side hasn't yet set up any listeners for the `onReanimatedPropsChange` event. Gut feeling says it doesn't cause any harm, but will continue to dig to make sure it doesn't

hannojg committed:
remove patch use fork

Santhosh-Sellavel commented:
@hannojg Can you please resolve conflicts?

hannojg committed:
Merge branch 'main' of github.com:margelo/expensify-app-fork into perf/rea-3-upgrade

hannojg committed:
install alpha1 version with reanimated fix

hannojg commented:
Conflicts resolved

Santhosh-Sellavel commented:
@hannojg I'm seeing this warning a lot on this branch

<img width="800" alt="Screenshot 2022-10-21 at 11 46 18 PM" src="https://user-images.githubusercontent.com/85645967/197262684-79702f49-06eb-4adf-9360-aeff4380ef34.png">

roryabraham commented:
> I also can't reproduce it all the time, was just seeing it once

I just logged into the app and it happened 🤷🏼 

> I'm seeing this warning a lot on this branch

That's concerning, as of [this PR](https://github.com/Expensify/App/pull/11550) we should no longer be using legacy drawer.

JmillsExpensify commented:
Are you all coming back to this PR today? Looks like we're at the finish line, just need to resolve the console errors. Right?

hannojg commented:
Yes, gonna look into it again today so we can merge it today!

hannojg commented:
Adding to my testing notes:

Warnings are reproducible when navigating from list to report, so its very likely coming from the drawer's REA usage

hannojg commented:
Okay @everyone, we can safely ignore this warning. For two reasons:

- 1. Its just a developer warning that react native generated for us to let us know we can reduce sending events to the JS side, as no listerers for native events are set up yet. In REA 3 they forget a check for that, I will make a PR to get rid of that warning (however this doesn't have to block us from emerging IMO (but not sure how you guys treat new warnings in the console))

- 2. **Most importantly:** once we turn on the new arch, the code in REA that sends these events will be handled differently anyways, so we are not getting hit by those events:
<img width="1355" alt="Screenshot 2022-10-25 at 10 23 29" src="https://user-images.githubusercontent.com/16821682/197723131-353bc64d-45bb-403e-b75d-653d1cb34a07.png">

Will look into the web warnings shortly, but just quickly looking at it it just seems like a general warning that we can't really do anything about. We are not using the legacy code, however, the warnings are still generated just by the pure presence of the legacy code. I think this can/will only be resolved when the legacy implementation gets removed from the drawer lib. But maybe we can at least suppress these warnings if we want to.

hannojg commented:
REA issue and accompanying PR is up: https://github.com/software-mansion/react-native-reanimated/issues/3716

hannojg commented:
We can also ignore the web warnings. If you want to I can add a commit to suppress the warnings. Can only go away once the legacy code gets removed :) 

So as a summary I think we are good to merge

hannojg committed:
add REA patch, see:

https://github.com/software-mansion/react-native-reanimated/pull/3717

hannojg committed:
run patch-package in post install

roryabraham commented:
Created [this issue](https://github.com/Expensify/App/issues/12118) to follow-up and remove the patch

roryabraham commented:
Sorry @hannojg, I think you need to reinstall pods and commit the diff

hannojg committed:
update package-lock file

hannojg committed:
update Podfile.lock

hannojg committed:
apply changes from REA iOS install fix

https://github.com/software-mansion/react-native-reanimated/pull/3701

Only applied parts, because on REA's main branch the ios project setup changed a bit so i needed to only use whats necessary for the current setup.\

This part of the patch can be reverted once the PR landed and we updated the REA version

hannojg committed:
updated podspec after patch

Santhosh-Sellavel commented:
I am seeing changes committed frequently. @hannojg is this ready for review?

AndrewGable commented:
Yes it is, it has some priority so if you could review ASAP @Santhosh-Sellavel that would be great!

AndrewGable reviewed (approved):
## Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

## Screenshots
<img width="1519" alt="Screen Shot 2022-10-25 at 11 28 40 AM" src="https://user-images.githubusercontent.com/2838819/197843568-1dc9eb72-1aba-409f-8a12-73962e477fcd.png">
<img width="506" alt="Screen Shot 2022-10-25 at 11 31 39 AM" src="https://user-images.githubusercontent.com/2838819/197843575-e8944707-29e9-4be0-bee4-abdf239720da.png">
<img width="564" alt="Screen Shot 2022-10-25 at 11 34 33 AM" src="https://user-images.githubusercontent.com/2838819/197843577-79c62dc7-432d-4652-bcad-9c7f387737f1.png">
<img width="1312" alt="Screen Shot 2022-10-25 at 11 36 09 AM" src="https://user-images.githubusercontent.com/2838819/197843583-06b915b0-d15d-4ec9-8afd-50f8fecc9a29.png">
<img width="747" alt="Screen Shot 2022-10-25 at 11 25 01 AM" src="https://user-images.githubusercontent.com/2838819/197843589-64c4d5a5-801b-47bf-9a41-410410725ca6.png">

----

@Santhosh-Sellavel - Let me know when you've approved and I can merge 🚀

Santhosh-Sellavel commented:
@AndrewGable We are Ignoring the console errors, right?

AndrewGable commented:
From reading https://github.com/Expensify/App/pull/11567#issuecomment-1290426741 - Yes, I believe we are OK to ignore warnings.

Santhosh-Sellavel commented:
<details>
<Summary>
Screen Recording
</Summary>

#### Desktop
https://user-images.githubusercontent.com/85645967/197847532-d324246f-622a-4be0-bb8a-baa46f68a94f.mov

#### Web
https://user-images.githubusercontent.com/85645967/197848208-4467d878-ccb7-49e5-83a2-3ae1708f7009.mov

#### iOS (Build using Xcode)
https://user-images.githubusercontent.com/85645967/197848247-8d4aa805-e88b-4641-9259-548f76c86b99.mp4

#### Android
https://user-images.githubusercontent.com/85645967/197848307-deeadd20-1db3-4944-9d97-b8979b4f8f16.mov

#### mWeb
https://user-images.githubusercontent.com/85645967/197848837-02ccc341-7207-4371-b4e3-08315ea289ae.mp4
</details>

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

Santhosh-Sellavel commented:
@AndrewGable I am having trouble while running yarn ios, build always fails does that work for you? I've used Xcode app to build the app.

AndrewGable commented:
I am able to successfully build via `npm run ios`

Santhosh-Sellavel commented:
Thanks then we are good to merge!

Santhosh-Sellavel reviewed (approved):
https://github.com/Expensify/App/pull/11567#issuecomment-1290976856

Santhosh-Sellavel commented:
iOS build working now after `npm run clean`
#### iOS 
https://user-images.githubusercontent.com/85645967/197873248-c5c7fb96-f0cd-4aa5-8933-1400c9f568b8.mp4

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/3332937820) to staging by @AndrewGable in version: 1.2.20-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/3342265019) to production by @chiragsalian in version: 1.2.20-3 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
